components/InputBox/index.tsx

```diff
@@ -83,6 +285,236 @@ const InputBox = ({
+              buttonNeutral: 'Ask Me Later',
+              buttonNegative: 'Cancel',
+              buttonPositive: 'OK',
+            },
+          );
+          if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+            return true;
+          } else if (granted === PermissionsAndroid.RESULTS.NEVER_ASK_AGAIN) {
+            Alert.alert(
+              'Storage Permission Required',
+              'App needs access to your storage to read files. Please go to app settings and grant permission.',
+              [
+                {text: 'Cancel', style: 'cancel'},
+                {text: 'Open Settings', onPress: Linking.openSettings},
+              ],
+            );
+          } else {
+            return false;
+          }
+        } catch (err) {
+          return false;
+        }
+      } else {
+        try {
+          const grantedImageStorage = await PermissionsAndroid.requestMultiple([
+            PermissionsAndroid.PERMISSIONS.READ_MEDIA_IMAGES,
+            PermissionsAndroid.PERMISSIONS.READ_MEDIA_VIDEO,
+          ]);
+          if (
+            grantedImageStorage['android.permission.READ_MEDIA_IMAGES'] &&
+            grantedImageStorage['android.permission.READ_MEDIA_VIDEO'] ===
+              PermissionsAndroid.RESULTS.GRANTED
+          ) {
+            return true;
+          } else if (
+            grantedImageStorage['android.permission.READ_MEDIA_IMAGES'] ===
+              PermissionsAndroid.RESULTS.NEVER_ASK_AGAIN ||
+            grantedImageStorage['android.permission.READ_MEDIA_VIDEO'] ===
+              PermissionsAndroid.RESULTS.NEVER_ASK_AGAIN
+          ) {
+            Alert.alert(
+              'Storage Permission Required',
+              'App needs access to your storage to read files. Please go to app settings and grant permission.',
+              [
+                {text: 'Cancel', style: 'cancel'},
+                {text: 'Open Settings', onPress: Linking.openSettings},
+              ],
+            );
+            return false;
+          } else {
+            return false;
+          }
+        } catch (err) {
+          console.warn(err);
+          return false;
+        }
+      }
+    }
+  }
+
+  // function handles the selection of images and videos
+  const handleGallery = async () => {
+    if (Platform.OS === 'ios') {
+      selectGalley();
+    } else {
+      let res = await requestStoragePermission();
+      if (res === true) {
+        selectGalley();
+      }
+    }
+  };
+
+  // function handles the slection of documents
+  const handleDoc = async () => {
+    if (Platform.OS === 'ios') {
+      selectDoc();
+    } else {
+      let res = await requestStoragePermission();
+      if (res === true) {
+        selectDoc();
+      }
+    }
+  };
+
+  const uploadResource = async (selectedImages: any, conversationID: any) => {
+    if (isUploading) return;
+
+    for (let i = 0; i < selectedImages?.length; i++) {
+      let attachmentType = selectedImages[i]?.type?.split('/')[0];
```

Comment:
Why this code is here?
Move to this code to some util, and then use it chatroom or media picker.